lib-utilities/config/config.go

```diff
@@ -531,3 +542,34 @@ func checkConnectionMethodConf() error {
 	}
 	return err
 }
+
+func checkEventConf() error {
+	if Data.EventConf == nil {
+		log.Warn("EventConf not provided, setting default value")
+		Data.EventConf = &EventConf{
+			DeliveryRetryAttempts:                 DefaultDeliveryRetryAttempts,
+			DeliveryRetryIntervalSeconds:          DefaultDeliveryRetryIntervalSeconds,
+			RetentionOfUndeliveredEventsInMinutes: DefaultRetentionOfUndeliveredEventsInMinutes,
+		}
+		return nil
+	}
+	if Data.EventConf.DeliveryRetryAttempts <= 0 {
+		log.Warn("No value found for DeliveryRetryAttempts, setting default value")
+		Data.EventConf.DeliveryRetryAttempts = DefaultDeliveryRetryAttempts
+	}
+	if Data.EventConf.DeliveryRetryIntervalSeconds <= 0 {
+		log.Warn("No value found for DeliveryRetryIntervalSeconds, setting default value")
+		Data.EventConf.DeliveryRetryIntervalSeconds = DefaultDeliveryRetryIntervalSeconds
+	}
+	if SaveUndeliveredEventsFlag {
+		reattempt := Data.EventConf.DeliveryRetryAttempts * Data.EventConf.DeliveryRetryIntervalSeconds
+		if reattempt < (Data.EventConf.RetentionOfUndeliveredEventsInMinutes / 60) {
+			return fmt.Errorf("configured value can't be less than the delivery attempt values")
```

Comment:
Mention which configured value is cant be less

Reply:
updated